projects/plugins/jetpack/modules/comments/comments.php

```diff
@@ -479,104 +479,53 @@ public function watch_comment_parent() {
-				window.location.hash = window.location.hash;
-			}
-		</script>
-		<![endif]-->
 		<script type="text/javascript">
-			(function () {
-				var comm_par_el = document.getElementById( 'comment_parent' ),
-					comm_par = ( comm_par_el && comm_par_el.value ) ? comm_par_el.value : '',
-					frame = document.getElementById( 'jetpack_remote_comment' ),
-					tellFrameNewParent;
-
-				tellFrameNewParent = function () {
-					if ( comm_par ) {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>" + '&replytocom=' + parseInt( comm_par, 10 ).toString();
-					} else {
-						frame.src = "<?php echo esc_url_raw( $this->signed_url ); ?>";
-					}
-				};
-
-				<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
-
-				if ( 'undefined' !== typeof addComment ) {
+			<?php if ( get_option( 'thread_comments' ) && get_option( 'thread_comments_depth' ) ) : ?>
+			const iframe = document.querySelector( '#jetpack_remote_comment' );
+			const watchReply = function() {
+				// Check addComment._Jetpack_moveForm to make sure we don't monkey-patch twice.
+				if ( 'undefined' !== typeof addComment && ! addComment._Jetpack_moveForm ) {
+					// Cache the Core function.
 					addComment._Jetpack_moveForm = addComment.moveForm;
-
-					addComment.moveForm = function ( commId, parentId, respondId, postId ) {
-						var returnValue = addComment._Jetpack_moveForm( commId, parentId, respondId, postId ),
-							cancelClick, cancel;
-
-						if ( false === returnValue ) {
-							cancel = document.getElementById( 'cancel-comment-reply-link' );
-							cancelClick = cancel.onclick;
-							cancel.onclick = function () {
-								var cancelReturn = cancelClick.call( this );
-								if ( false !== cancelReturn ) {
-									return cancelReturn;
-								}
-
-								if ( ! comm_par ) {
-									return cancelReturn;
-								}
-
-								comm_par = 0;
-
-								tellFrameNewParent();
-
-								return cancelReturn;
-							};
+					const commentParent = document.querySelector( '#comment_parent' );
+					const cancel = document.querySelector( '#cancel-comment-reply-link' );
+
+					function tellFrameNewParent ( commentParentValue ) {
+						const url = new URL(iframe.src);
+						if ( commentParentValue ) {
+							url.searchParams.set( 'replytocom', commentParentValue )
+						} else {
+							url.searchParams.delete( 'replytocom' );
 						}
-
-						if ( comm_par == parentId ) {
-							return returnValue;
+						if( iframe.src !== url.href ) {
+							iframe.src = url.href;
 						}
+					};
 
-						comm_par = parentId;
-
-						tellFrameNewParent();
+					cancel.addEventListener('click', function () {
+						tellFrameNewParent( false );
+					});
 
-						return returnValue;
+					addComment.moveForm = function ( commId, parentId, respondId, postId ) {
+						tellFrameNewParent( parentId );
+						return  addComment._Jetpack_moveForm( commId, parentId, respondId, postId )
 					};
```

Comment:
When I was testing this I noticed that `addComment.moveForm` actually takes five parameters, not just four. Omitting the last one results in the heading always being "Leave a reply" rather than changing to "Reply to Whoever" like core does.

This is an existing bug, but since we're rewriting all this anyway it's easy to fix:
```suggestion
					addComment.moveForm = function ( commId, parentId, respondId, postId, replyTo ) {
						tellFrameNewParent( parentId );
						return  addComment._Jetpack_moveForm( commId, parentId, respondId, postId, replyTo )
					};
```
Or we could do like this:
```suggestion
					addComment.moveForm = function ( ...args ) {
						tellFrameNewParent( args[1] );
						return  addComment._Jetpack_moveForm( ...args )
					};
```

(I also noticed another already-existing bug, Jetpack's form doesn't get focused like core's does when you click a "Reply" link, but that seems harder to fix so should probably be a separate PR)

Reply:
Thank you so much. I addressed this.